Src/IronPython/Runtime/LiteralParser.cs

```diff
@@ -212,19 +186,31 @@ public static class LiteralParser {
                 }
             }
             return buf?.ToString();
+
+            void handleError(int start, int end, string reason) {
+                bytesData ??= data is byte[] ba ? new Bytes(ba) : Bytes.Empty;
+
+                if (errorHandler == null || start >= bytesData.Count) {
```

Comment:
If `data` is not `byte[]` it will ignore `errorHandler` and always throw (since `bytesData.Count == 0`). Is this the expected behavior? If we expect `errorHandler` to always be `null` with `char[]` then I would add a guard (`Debug.Assert` or `NotImplementedException`) at the top of the method.

Reply:
Yes, it was the intended behaviour. The current implementation assumes `errorHandler` to be always `default` with `char[]`, since currently there is no need for a non-default handler in such case. If in the future this assumption is no longer valid, `handleError` has to be updated. I agree that a guard with `NotImplementedException` is better than a silent ignore.